parsers/src/test/scala/uk/gov/ons/addressIndex/crfscala/CrfFeatureTest.scala

```diff
@@ -0,0 +1,120 @@
+package uk.gov.ons.addressIndex.crfscala
+
+import org.scalatest.{FlatSpec, Matchers}
+import uk.gov.ons.addressIndex.crfscala.CrfFeatureAnalyser.CrfFeatureAnalyser
+import scala.util.Try
+
+class CrfFeatureTest extends FlatSpec with Matchers  {
+
+  object TestInstanceBoolean {
+    type tType = Boolean
```

Comment:
@iRhysBradbury I agree with @Mironor that this should be type classes, not duplicated. I'd expect to see a `trait TestInstance[TypeParam]` which you could simply extend, maybe even with implicit type class instances that you could materialize as context bounds.


```scala
trait TestInstance[T] {
 // ....
 def pullKateUpton(name: String, ambition: NumericRange[Long]): Try[T] = ...
}

object TestInstance {
   // this is just a convenience method.
   def apply[T : TestInstance]: TestInstance[T] = implicitly[TestInstance[T]]

   // it's equivalent to this basically
   def apply[T]()(implicit ev: TestInstance[T]): TestInstance[T] = ev

   implicit object TestInstanceBool extends TestInstance[Boolean] {
     override def pullKateUpton(name: String, ambition: NumericRange[Long]: Try[Boolean] = {
       if ((name == "Flavian" || name == "Rhys") && ambition.contains(10L))  {
          Success(true)
       } else {
          throw new Error("How dare you even try"))
       }
     }
   }
}
```
// now this is pretty fun since it works like magic.

```scala
val testInstance = TestInstance[Boolean]

testInstance.pullKateUpton("rhys", 1 to 5L) // false
testInstance.pullKateUpton("rhys", 1 to 10L) // TRUE !!! send pics please
```